exporter/opensearchexporter/trace_exporter.go

```diff
@@ -0,0 +1,99 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package opensearchexporter contains an opentelemetry-collector exporter
+// for OpenSearch.
+package opensearchexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/opensearchexporter"
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
+)
+
+type opensearchTracesExporter struct {
+	logger *zap.Logger
+
+	index       string
+	maxAttempts int
+
+	client      *osClientCurrent
+	bulkIndexer osBulkIndexerCurrent
+	model       mappingModel
+}
+
+func newTracesExporter(logger *zap.Logger, cfg *Config) (*opensearchTracesExporter, error) {
+	if err := cfg.Validate(); err != nil {
+		return nil, err
+	}
+
+	client, err := newOpenSearchClient(logger, cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	bulkIndexer, err := newBulkIndexer(logger, client, cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	maxAttempts := 1
+	if cfg.Retry.Enabled {
+		maxAttempts = cfg.Retry.MaxRequests
+	}
+
+	// TODO: Apply encoding and field mapping settings.
+	model := &encodeModel{dedup: true, dedot: false}
+
+	return &opensearchTracesExporter{
+		logger:      logger,
+		client:      client,
+		bulkIndexer: bulkIndexer,
+
+		index:       cfg.TracesIndex,
+		maxAttempts: maxAttempts,
+		model:       model,
+	}, nil
+}
+
+func (e *opensearchTracesExporter) Shutdown(ctx context.Context) error {
+	return e.bulkIndexer.Close(ctx)
+}
+
+func (e *opensearchTracesExporter) pushTraceData(
+	ctx context.Context,
+	td ptrace.Traces,
+) error {
+	var errs []error
+	resourceSpans := td.ResourceSpans()
+	for i := 0; i < resourceSpans.Len(); i++ {
+		il := resourceSpans.At(i)
+		resource := il.Resource()
+		scopeSpans := il.ScopeSpans()
+		for j := 0; j < scopeSpans.Len(); j++ {
+			spans := scopeSpans.At(j).Spans()
+			for k := 0; k < spans.Len(); k++ {
+				if err := e.pushTraceRecord(ctx, resource, spans.At(k)); err != nil {
+					if cerr := ctx.Err(); cerr != nil {
+						return cerr
+					}
+					errs = append(errs, err)
+				}
+			}
+		}
+	}
+
+	return multierr.Combine(errs...)
+}
+
+func (e *opensearchTracesExporter) pushTraceRecord(ctx context.Context, resource pcommon.Resource, span ptrace.Span) error {
```

Comment:
Do you plan adding support for service maps as well?

Reply:
Service map is not planned currently in this exporter - we are thinking of using the existing OTEL contribute service graph processor and consolidate that functionality with data prepper